### PR TITLE
feat(charmil-crud): add validations and flag completion

### LINKED ISSUE #179: Add validation for flags in Charmil CRUD generator templates

### PR DESCRIPTION
**Pull request by namit-chandwani**
Closes #179 

### Description
Made the following changes:
- Added validations for flags and arguments of CRUD commands
- Added flag completion functionality

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [x] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by namit-chandwani** [comment]
This PR is now ready for review.
Please ignore the failing build as it's related to issue #199 and I'll need to fix it in a follow-up PR shortly after this one gets merged.

/cc @aerogear/charmil

File: cli/internal/template/crud/create/create.go [hunk 1]
```diff
@@ -4,27 +4,46 @@ package create
 
 import (
 	"github.com/aerogear/charmil/cli/internal/factory"
+	"github.com/aerogear/charmil/cli/internal/flagutil"
```
Comment (ankithans) [inline cli/internal/template/crud/create/create.go:7]: @namit-chandwani i think you misplaced the imports. It should be from starter?

File: cli/internal/template/crud/create/create.go [hunk 2]
```diff
@@ -4,27 +4,46 @@ package create
...
+	validNameRegexp string
 
 	// You can add more fields here according to your requirements
 }
 
 // GetCreateCommand returns a new command for creating {{.Singular}} instances.
 func GetCreateCommand(f *factory.Factory) *cobra.Command {
-	opts := &createOptions{}
+	opts := &createOptions{
+		validNameRegexp: `^[a-z]([-a-z0-9]*[a-z0-9])?$`,
```
Comment (ankithans) [inline cli/internal/template/crud/create/create.go:30]: @wtrocki In the UseMatches rule(Validator), shall we do regex validation for flags as well? Currently we do this for `Use` only.

File: cli/internal/template/crud/create/create.go (hunk 1 above)
Comment (namit-chandwani) [reply cli/internal/template/crud/create/create.go:7]: Yes, I did that on purpose. Mentioned it over here: https://github.com/aerogear/charmil/pull/204#issuecomment-894050185

Once this PR gets merged, I'll create a new one that fixes issue #199 (ie. convert all compilable go files into templates and update imports)

**Review by ankithans — APPROVED** [review]

**Comment by namit-chandwani** [comment]
Creating a PR to the `charmil-starter` repo now, which will add the `flagutil` package over there too.

**Comment by namit-chandwani** [comment]
@wtrocki I'm creating a follow-up PR in some time which would fix issue #199. 
So I think it would be better if we make a release after that is merged

**Pull request merged**